File: angular/src/app/models/user-status-dto.ts
import { CompareOperation } from "@shared/AppEnums";

export interface UserStatusDto {
  id: string;
  displayName: string;
  level: number;
  lowCompareOperation: CompareOperation;
  requiredNumber: number;
}

export interface PagedResultDto {
  totalCount: number | undefined;
  items: UserStatusDto[] | undefined;
}

export interface Result {
  result: PagedResultDto;
}

export interface ResultObject {
  result: UserStatusDto;
}

File: aspnet-core/src/RMALMS.Application/UserStatus/Dto/UserStatusDto.cs
using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using RMALMS.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace RMALMS.UserStatus.Dto
{
    [AutoMapTo(typeof(Entities.UserStatus))]
    public class UserStatusDto: EntityDto<Guid>
    {
        public int Level { get; set; }
        public string DisplayName { get; set; }
        public bool IsStatic { get; set; }
        public CompareOperation? LowCompareOperation { get; set; }
        public int? RequiredNumber { get; set; }
    }
}
